core/ui/common/qam/powertools_menu.gd

```diff
@@ -0,0 +1,316 @@
+extends VBoxContainer
+
+var Platform := load("res://core/global/platform.tres")
+
+const AMD_GPU_MIN_MHZ: float = 200
+const POWERTOOLS_PATH : String = "/usr/share/opengamepadui/scripts/powertools"
+
+@onready var performance_manager := load("res://core/systems/performance/performance_manager.tres") as PerformanceManager
+@onready var launch_manager := load("res://core/global/launch_manager.tres") as LaunchManager
+
+var command_timer: Timer
+var update_timer: Timer
+
+@onready var cpu_boost_button := $CPUBoostButton
+@onready var cpu_cores_slider := $CPUCoresSlider
+@onready var gpu_freq_enable := $GPUFreqButton
+@onready var gpu_freq_max_slider := $GPUFreqMaxSlider
+@onready var gpu_freq_min_slider := $GPUFreqMinSlider
+@onready var gpu_temp_slider := $GPUTempSlider
+@onready var power_profile_dropdown := $PowerProfileDropdown
+@onready var tdp_boost_slider := $TDPBoostSlider
+@onready var tdp_slider := $TDPSlider
+@onready var thermal_profile_dropdown := $ThermalProfileDropdown
+@onready var smt_button := $SMTButton
+
+@onready var platform : PlatformProvider = Platform.platform
+
+var logger := Log.get_logger("PowerTools", Log.LEVEL.INFO)
+
+
+# Called when the node enters the scene tree for the first time.
+# Finds default values and current settings of the hardware.
+func _ready() -> void:
+	await performance_manager.read_system_components()
+	_setup_interface()
+	performance_manager.load_profile()
+	launch_manager.app_switched.connect(_on_app_switched)
+
+	command_timer = Timer.new()
+	command_timer.set_autostart(false)
+	command_timer.set_one_shot(true)
+	add_child(command_timer)
+
+
+func _setup_interface() -> void:
+	if performance_manager.cpu.smt_capable:
+		logger.debug("CPU is SMT Capable")
+		_setup_cpu_core_range()
+	if performance_manager.cpu.boost_capable:
+		logger.debug("CPU is Boost Capable")
+		_setup_cpu_boost()
+	if performance_manager.gpu.tdp_capable:
+		logger.debug("SOC is TDP Capable")
+		_setup_tdp_range()
+	if performance_manager.gpu.clk_capable:
+		logger.debug("GPU is Reclock Capable")
+		_setup_gpu_freq_range()
+	if performance_manager.gpu.power_profile_capable:
+		logger.debug("GPU is Power Profile Capable")
+		_setup_power_profile()
+	if performance_manager.gpu.tj_temp_capable:
+		logger.debug("GPU is TJ Temp Configurable")
+		_setup_gpu_temp()
+	if performance_manager.gpu.thermal_mode_capable:
```

Comment:
The `thermal_mode_capable` property does not exist on the `GPUInfo` class.

Reply:
SAA